extensions/data-plane/data-plane-api/src/main/java/org/eclipse/edc/connector/dataplane/api/DataPlaneApiExtension.java

```diff
@@ -102,20 +99,25 @@ public void initialize(ServiceExtensionContext context) {
 
         webService.registerResource(controlApiConfiguration.getContextAlias(), new DataPlaneControlApiController(dataPlaneManager));
 
-        configuration = webServiceConfigurer.configure(context, webServer, PUBLIC_SETTINGS);
+        var publicApiConfig = publicApiConfig(context);
         var publicApiController = new DataPlanePublicApiController(dataPlaneManager, dataAddressResolver, monitor, executorService);
-        webService.registerResource(configuration.getContextAlias(), publicApiController);
+        webService.registerResource(publicApiConfig.getContextAlias(), publicApiController);
     }
 
-    @Provider
-    public DataPlanePublicApiUrl dataPlanePublicApiUrl() {
+    @Provider(isDefault = true)
+    public DataPlanePublicApiUrl dataPlanePublicApiUrl(ServiceExtensionContext context) {
         try {
-            var url = new URL(String.format("http://%s:%s%s", hostname.get(), configuration.getPort(), configuration.getPath()));
+            var config = publicApiConfig(context);
```

Comment:
the `webServiceConfigurer` could be `null` by the time the default provider method is invoked.

Reply:
> I do not quite understand why this could not be done through configuration?

`DataPlanePublicApiUrl` removed as discussed